server/interview/interview.service.js

```diff
@@ -0,0 +1,39 @@
+const db = require('../helpers/db');
+
+const questionService = require('../question/question.service');
+const userService = require('../users/user.service');
+
+const Interview = db.Interview;
+
+module.exports = {
+    createInterview,
+    updateInterview
+}
+
+async function createInterview(userId, difficulty) {
+    let interview = new Interview();
+    let participant = await userService.getById(userId);
+    let question = await questionService.findARandomQuestionByDifficulty(difficulty);
+    interview.participants.push(participant);
+    interview.questions.push(question);
+    await interview.save();
+
+    return interview;
+}
+
+async function updateInterview(userId, difficulty, interviewId) {
+    let updatedInterview = await Interview
+        .findById(interviewId)
+        .populate('participants')
+        .populate('questions');
```

Comment:
I don't think you need to populate here - this makes extra unnecessary API calls to the database, much slows down performance

Reply:
Yes it shouldn't be here. I removed the calls.